src/features/joinForms/panes/JoinFormPane.tsx

```diff
@@ -0,0 +1,132 @@
+import { FC } from 'react';
+import {
+  Autocomplete,
+  Box,
+  IconButton,
+  TextField,
+  useTheme,
+} from '@mui/material';
+
+import { Delete } from '@mui/icons-material';
+import globalMessageIds from 'core/i18n/globalMessageIds';
+import messageIds from '../l10n/messageIds';
+import { NATIVE_PERSON_FIELDS } from 'features/views/components/types';
+import PaneHeader from 'utils/panes/PaneHeader';
+import useCustomFields from 'features/profile/hooks/useCustomFields';
+import useJoinForm from '../hooks/useJoinForm';
+import useJoinFormMutations from '../hooks/useJoinFormMutations';
+import { useMessages } from 'core/i18n';
+
+type Props = {
```

Comment:
I tend to disagree, and I will explain why. This is a locally defined type. It's not to be shared with the outside world. It is just defined here, and then immediately used on the next line, and not exported or used anywhere else.

In that sense, it resembles a local variable. For example, if you have a state inside of a checkbox component that indicates whether that component is selected, you would probably name it `selected` and not `checkboxSelected` (because the variable exists locally inside a checkbox component and everything in that component is related to the checkbox unless otherwise explicitly stated).

I could think of other examples as well, but I'm sure you can too so I don't think that's necessary. 

I don't like the really long names for single-use, private/local types because all that they do is make the code more verbose, which at least to me doesn't make it more understandable. I'm sure there are other cases where I would come to a different conclusion, but for components "props" is such a fundamental concept that I think it should be enough to just call them `Props` and people would understand that they are the props of the component whose file you've just opened.